python/ray/data/datasource/file_meta_provider.py

```diff
@@ -194,14 +200,21 @@ class FastFileMetadataProvider(DefaultFileMetadataProvider):
     negligible for local filesystems, it can be substantial for cloud storage service
     providers.
 
-    This should only be used when all input paths are known to be files.
+    This should only be used when all input paths exist and are known to be files.
     """
 
     def expand_paths(
         self,
         paths: List[str],
         filesystem: "pyarrow.fs.FileSystem",
+        ignore_missing_paths: bool = False,
     ) -> Tuple[List[str], List[Optional[int]]]:
+        if ignore_missing_paths:
+            raise ValueError(
+                "`ignore_missing_paths` cannot be set when used with "
+                "`FastFileMetadataProvider`. All paths must exist when "
+                "using `FastFileMetadataProvider`."
+            )
```

Comment:
I removed the check in `file_based_datasource`. Missing files are identified only during metadata checking, to prevent metadata and the actual files from differing.

Reply:
Cool, thanks!

Reply:
It seems like we could catch the `NotFound` exception raised by `fs.open_input_stream()` or however they propagate not found errors and drop the path, where the corresponding block metadata may end up reflecting an empty block. Definitely some things to figure out there, so saving that as a follow-up sounds good.